dev/Repeater/ElementManager.cpp

```diff
@@ -270,8 +270,36 @@ void ElementManager::DataSourceChanged(const winrt::IInspectable& /*source*/, wi
 
         case winrt::NotifyCollectionChangedAction::Replace:
         {
-            OnItemsRemoved(args.OldStartingIndex(), args.OldItems().Size());
-            OnItemsAdded(args.NewStartingIndex(), args.NewItems().Size());
+            int oldSize = args.OldItems().Size();
+            int newSize = args.NewItems().Size();
+            int oldStartIndex = args.OldStartingIndex();
+            int newStartIndex = args.NewStartingIndex();
+
+            if (oldSize == newSize &&
+                oldStartIndex == newStartIndex &&
+                IsDataIndexRealized(oldStartIndex) &&
```

Comment:
Is this check necessary? If x of the replaced y items would be at the end or at the beginning, could we still clear those x elements?

Reply:
This was a simple check to add. I will try to go through all the cases here where a subset of the elements overlap the realized range.